src/System.Windows.Forms/src/System/Windows/Forms/Controls/DataGridView/DataGridView.Methods.cs

```
@@ -29400,41 +29417,62 @@ private void UpdateMouseEnteredCell(HitTestInfo hti, MouseEventArgs e)
             &&  oldEdgeRowIndex == newEdgeRowIndex)
         {
             // h1
-            DataGridViewColumn dataGridViewColumn = Columns.GetNextColumn(Columns[oldEdgeColumnIndex], DataGridViewElementStates.Visible, DataGridViewElementStates.None);
+            DataGridViewColumn? dataGridViewColumn = Columns.GetNextColumn(Columns[oldEdgeColumnIndex], DataGridViewElementStates.Visible, DataGridViewElementStates.None);
             Debug.Assert(dataGridViewColumn is not null);
```

I see that sometimes we are doing this `Debug.Assert` and sometimes checking `dataGridViewColumn` with an if statement to see if it is null. It seems that we are not expecting `dataGridViewColumn` to be null across this method where it is defined, so can you change to throw an `InvalidOperationException` instead? 

If we have a lot of instances where we are expecting non-null `DataGridViewColumn` on calls to `GetNextColumn/GetPreviousColumn` perhaps consider adding an internal helper method on `DataGridViewColumnCollection` like `GetNextColumnRequired/GetPreviousColumnRequired`. Otherwise a small local helper method here would be ok.